stl/inc/chrono

```diff
@@ -2086,6 +2087,689 @@ namespace chrono {
+            {sys_seconds{seconds{1341100800}}, true},
+            {sys_seconds{seconds{1435708800}}, true},
+            {sys_seconds{seconds{1483228800}}, true},
+        };
+        constexpr size_t _Pre_2018_count = _STD size(_Pre_2018_leap_seconds);
+
+        size_t _New_reg_ls_size;
+        auto _Reg_ls_data = unique_ptr<_RegistryLeapSecondInfo[]>{__std_tzdb_get_reg_leap_seconds(
+            (_STD max)(_Current_size, _Pre_2018_count) - _Pre_2018_count, &_New_reg_ls_size)};
+
+        const size_t _New_size = _Pre_2018_count + _New_reg_ls_size;
+        if (_New_reg_ls_size > _Current_size && !_Reg_ls_data) {
+            _Xbad_alloc();
+        } else if (_New_reg_ls_size == 0 && _Reg_ls_data) {
+            _XGetLastError();
+        }
+
+        vector<leap_second> _Leap_sec_info;
+        bool _All_ls_positive = true;
+
+        if (_New_size > _Current_size) {
+            _Leap_sec_info.reserve(_New_size);
+            _Leap_sec_info.insert(
+                _Leap_sec_info.begin(), _STD cbegin(_Pre_2018_leap_seconds), _STD cend(_Pre_2018_leap_seconds));
+
+            for (size_t _Idx = 0; _Idx < _New_reg_ls_size; ++_Idx) {
+                // Leap second happens at ls._Hour:59:59. We store the next second after, so add a whole hour.
+                const auto& _Ls = _Reg_ls_data[_Idx];
+                const auto _Date =
+                    static_cast<sys_days>(year_month_day{year{_Ls._Year}, month{_Ls._Month}, day{_Ls._Day}})
+                    + hours{_Ls._Hour + 1};
+                _Leap_sec_info.emplace_back(_Date, !_Ls._Negative);
+                _All_ls_positive = _All_ls_positive && !_Ls._Negative;
+            }
+        }
+
+        return {_Leap_sec_info, _All_ls_positive};
+    }
+
+    // [time.zone.db]
+
+    // TRANSITION: work in progress
+    struct tzdb {
+        vector<leap_second> leap_seconds;
+        bool _All_ls_positive;
+    };
+
+    // TRANSITION: work in progress
+    class tzdb_list {
+    private:
+        using _ListType = forward_list<tzdb>;
+
+    public:
+        using const_iterator = _ListType::const_iterator;
+
+        tzdb_list(const tzdb_list&) = delete;
+        tzdb_list& operator=(const tzdb_list&) = delete;
+        tzdb_list()                            = default;
+
+        const tzdb& front() const noexcept {
+            _Shared_lock _Lk(_Tzdb_mutex);
+            return _Tzdb_list.front();
+        }
+
+        template <class... _ArgsTy>
+        void _Emplace_front(_ArgsTy&&... _Args) {
+            _Unique_lock _Lk(_Tzdb_mutex);
+            _Tzdb_list.emplace_front(_STD forward<_ArgsTy>(_Args)...);
+        }
+
+        const tzdb& _Reload() {
+            _Unique_lock _Lk(_Tzdb_mutex);
+            auto [_Leap_sec, _All_ls_positive] = _Xtzdb_generate_leap_seconds(_Tzdb_list.front().leap_seconds.size());
+            if (!_Leap_sec.empty()) {
+                _Emplace_front(tzdb{_STD move(_Leap_sec), _All_ls_positive});
+            }
+            return front();

```

Comment:
This is holding `_Unique_lock _Lk(_Tzdb_mutex);` and then calling `front()` which acquires `_Shared_lock _Lk(_Tzdb_mutex);`. Even if recursive locking is permitted, this seems inefficient. I believe we should directly return `_Tzdb_list.front()` here.